execution/commitment/hex_patricia_hashed.go

```diff
@@ -1197,6 +1197,108 @@ func (hph *HexPatriciaHashed) witnessCreateAccountNode(c *cell, row int, hashedK
 	return accountNode, nil
 }
 
+func readBranchData(hph *HexPatriciaHashed, key []byte, depth int16) ([16]*cell, error) {
+	var rowData [16]*cell
+	compactKey := hexNibblesToCompactBytes(key)
+	branchData, _, err := hph.ctx.Branch(compactKey)
+	if err != nil {
+		return rowData, err
+	}
+	if branchData == nil {
+		return rowData, fmt.Errorf("empty branch data for key %x", compactKey)
+	}
+	bd := BranchData(branchData)
+	_, _, rowData, err = bd.decodeCells()
+	if err != nil {
+		return rowData, err
+	}
+	return rowData, nil
+}
+
+// number of non-empty cells in a row
+func nCells(rowData [16]*cell) int {
+	var count int = 0
+	for i := 0; i < 16; i++ {
+		if !rowData[i].IsEmpty() {
+			count++
+		}
+	}
+	return count
+}
+
+// first index in row where the cell is not empty
+func firstNonEmptyIdx(rowData [16]*cell) int {
+	for i := 0; i < 16; i++ {
+		if !rowData[i].IsEmpty() {
+			return i
+		}
+	}
+	return -1
+}
+
+func terminalRowToFullNode(hph *HexPatriciaHashed, rowData [16]*cell, depth int16) (*trie.FullNode, error) {
+	var fullNode trie.FullNode
```

Comment:
tbh this function reminds me of `if chain` we have inside `toWitnessTrie`

Reply:
I've commented out this code now. Good to leave it there in case it's needed again.